src/main/java/mate/academy/repository/JdbcBookRepository.java

```diff
@@ -0,0 +1,119 @@
+        try (Connection conn = ConnectionUtil.getConnection();
+                PreparedStatement statement = conn.prepareStatement(sqlSelectById)) {
+            statement.setLong(1, id);
+            try (ResultSet rs = statement.executeQuery()) {
+                if (rs.next()) {
+                    return Optional.of(getBookFromResultSet(rs));
+                }
+                return Optional.empty();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find ", e);
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sqlSelectAll = "SELECT id, title, price FROM books";
+        try (Connection conn = ConnectionUtil.getConnection();
+                PreparedStatement statement = conn.prepareStatement(sqlSelectAll);
+                ResultSet rs = statement.executeQuery()) {
+            while (rs.next()) {
+                books.add(getBookFromResultSet(rs));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find all", e);
+        }
+        return books;
+    }
+
+    @Override
+    public Book updateById(Book book) {
+        String sqlUpdateById = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sqlUpdateById,
+                        PreparedStatement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+
+            checkIfUpdateSuccessful(statement.executeUpdate());
+        } catch (SQLException e) {
+            throw new RuntimeException("Could not update by id " + book.getId() + " ", e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String sqlDeleteById = "DELETE FROM books WHERE id = ?";
+        try (Connection conn = ConnectionUtil.getConnection();
+                PreparedStatement statement = conn.prepareStatement(sqlDeleteById)) {
+            statement.setLong(1, id);
+            return checkIfUpdateSuccessful(statement.executeUpdate());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Deleting by id " + id + " failed ", e);
+        }
+    }
+
+    private Book getBookFromResultSet(ResultSet resultSet) throws SQLException {
+        Book book = new Book();
+        book.setId(resultSet.getObject(COLUMN_ID, Long.class));
+        book.setTitle(resultSet.getString(COLUMN_TITLE));
+        book.setPrice(resultSet.getBigDecimal(COLUMN_PRICE));
+        return book;
+    }
+
+    private boolean checkIfUpdateSuccessful(int affectedRows) throws SQLException {
+        if (affectedRows <= 0) {
+            throw new SQLException("No affected rows");
```

Comment:
Are you sure that SQL exception should be thrown? Is it an SQL problem?

Reply:
> Are you sure that SQL exception should be thrown? Is it an SQL problem?

I see now.
I can throw Runtime exception or it is possible to throw DataProcessingException and catch it add description with what data and in what case it was thrown, and throw RuntimeException. It is much complicated. I try to commit with RuntimeException))))